packages/xrpc-server/src/server.ts

```diff
@@ -200,6 +240,91 @@ export class Server {
       }
     }
   }
+
+  protected async addSubscription(
+    nsid: string,
+    def: LexXrpcSubscription,
+    config: XRPCStreamHandlerConfig,
+  ) {
+    const assertValidXrpcParams = (params: unknown) =>
+      this.lex.assertValidXrpcParams(nsid, params)
+    const resolveLexUri = (ref: string) => this.lex.resolveLexUri(nsid, ref)
+    this.subscriptions.set(
+      nsid,
+      new XrpcStreamServer({
+        noServer: true,
+        handler: async function* (req) {
+          try {
+            const url = new URL(req.url || '', 'http://x')
```

Comment:
total nit, but can we move this down next to `decodeQueryParams` so its more obvious what its used for